src/features/devices/hooks/should-be-managed-by--release.ts

```diff
@@ -0,0 +1,111 @@
+import { hooks, sbvrUtils } from '@balena/pinejs';
+import { getAffectedIds } from '@balena/pinejs/out/sbvr-api/sbvr-utils';
+
+hooks.addPureHook('PATCH', 'resin', 'device', {
+	PRERUN: async (args) => {
+		const { request } = args;
+		const deviceIds = await getAffectedIds(args);
+
+		if (
+			request.values.should_be_managed_by__release != null ||
+			deviceIds.length === 0
+		) {
+			return;
+		}
+
+		/**
+		 * These device(s) should have a valid release assigned based on their current
+		 * running supervisor version, if we have one...
+		 */
+
+		const [device] = await args.api.get({
```

Comment:
the comment above seems to indicate this type of PATCH could include multiple devices, but here is handled as a single device. is that possible here?

Reply:
yep, good point.  So, since there are multiple devices, are we guaranteed that they will all need the same supervisor release id for `should_be_managed_by__release`?  In other words, each device might have a different `device.supervisor_version` and a different `device.is_for__device_type.is_of__cpu_architecture.id`, right?  So therefore, I can't just set `request.values.should_be_managed_by__release = X` and patch them all to the same release id.

Reply:
@xginn8 hey Matt, just pinging on this :point_up_2: 

Since there are multiple devices, the requirements around `should_be_managed_by__release` might be different for each one, right?  We can't just `request.values.should_be_managed_by__release = X` but rather have to call `api.patch(...)` for each device and patch it according to the correct release id.  Am I on the right track?

Reply:
Here's the current implementation in our closed source API:

https://github.com/balena-io/balena-api/blob/2efc36726e9f9ae93f40e8af5eec193dcac5d7c3/src/models/hooks/resources/device.ts#L347-L363
https://github.com/balena-io/balena-api/blob/2efc36726e9f9ae93f40e8af5eec193dcac5d7c3/src/models/hooks/resources/device.ts#L665-L733

Basically we map over devices grouped by CPU architecture and PATCH the values N architecture times instead of M devices times.

I think that should help clear up the ambiguity, sorry for my tardiness here!!